### ISSUE TITLE
[MEDIUM] Zulucrypt-gui 6.2.0: When entering a wrong password in zulucrypt-gui, the pane "Unlock encrypted volume" cannot be closed

### ISSUE THREAD

**Issue by odiep2** [mhogomchungu/zuluCrypt#222]
[MEDIUM] Zulucrypt-gui 6.2.0: When entering a wrong password in zulucrypt, the pane "Unlock encrypted volume" cannot be closed

[CrossPost]

I cross posted this issue on https://forums.linuxmint.com/viewtopic.php?p=2553418. Sorry, won't happen again.

[Problem]

- I boot [linuxmint-22-cinnamon-64bit.iso] (NOT installed, NOT customized), using Ventoy boot loader 1.0.99, installed on internal SSD
- I issue command [sudo apt install zulucrypt-gui]
- After I start zulucrypt-gui, I perform the following actions:

> [menu open\volume hosted in a file] > 
- The "Unlock encrypted Volume" pane is opened

> click icon behind "Volume path"
> select a Veracrypt-container > [Open]
> mount name = VeraCryptContainer123456.vcc_1
> off= Mount in read only mode
> off = Share Mount Point
> Volume type=Veracrypt
> PIM Value field = <empty>
> Pull-down-menu above the password field = Password (default)
> Password={wrongPassword}

- I enter a WRONG password in the password field

> Click [Open]

- A message pane with title "Unlock encrypted Volume" is shown, with the message: "Please be patient ... Elapsed time: {time}". There's no button to click in this pane, except for the [X] close button.

 (info) Normally, the 25GB Veracrypt container opens in the File Manager in about 8 seconds.

> When the Elapsed time is about 5 minutes, I click the [X] close button

- Nothing happens. The message pane "Unlock encrypted Volume", and the underlying GUI "Unlock encrypted Volume" cannot be closed, using buttons in the GUI.

- I have to perform the following actions to stop zulucrypt-gui:

> Go to [start menu\...\System Monitor\Processes] > 
    - zulucrypt-gui > terminate > [end process]
    - zulucrypt-cli > terminate > [end process]

[Cause]

Unknown

[Solution]

Unknown

[Workaround]

> Go to [LM-menu\administration\System Monitor\Processes] > 
    - zulucrypt-gui > terminate > [end process]
    - zulucrypt-cli > terminate > [end process]

> Start zulucrypt-gui again
> enter the required data
> Type the correct password in the field "Password"
- the Veracrypt container is opened correctly

[Information]

Packages:

> libzulucrypt-exe1.2.0/noble,now 6.2.0-1ubuntu3 amd64 
> libzulucrypt1.2.0/noble,now 6.2.0-1ubuntu3 amd64 
> libzulucryptpluginmanager1.0.0/noble,now 6.2.0-1ubuntu3 amd64
> zulucrypt-cli/noble,now 6.2.0-1ubuntu3 amd64 
> zulucrypt-gui/noble,now 6.2.0-1ubuntu3 amd64 
> zulupolkit/noble,now 6.2.0-1ubuntu3 amd64 

INXI:

> Host: mint Kernel: 6.8.0-38-generic arch: x86_64 bits: 64 compiler: gcc
>    v: 13.2.0 clocksource: tsc avail: hpet,acpi_pm
>    parameters: BOOT_IMAGE=/casper/vmlinuz boot=casper
 >   initrd=/casper/initrd.lz username=mint hostname=mint quiet splash --
>  Desktop: Cinnamon v: 6.2.7 tk: GTK v: 3.24.41 wm: Muffin v: 6.2.0 tools:
>    avail: cinnamon-screensaver vt: 7 dm: LightDM v: 1.30.0
>  Distro: Linux Mint 22 Wilma base: Ubuntu 24.04 noble

**Comment by mhogomchungu** [comment]
1. Enter a wrong password.
2. Enter a PIM value of "1".
3. Click "Open" and see if you will get a "wrong password" error within a minute or two.

**Comment by mhogomchungu** [comment]
There is no bug here.

VeraCrypt volumes supports a very long list of key combinations and all of them are tried when you enter a wrong password. Default behavior of checking for passwords is very expensive and trying all key combinations will take a very long time if a wrong password is entered and the computer is not very powerful.

If this delay bothers you then a solution is to create another volume that uses a PIM value that is low.

**Comment by odiep2** [comment]
Dear @mhogomchungu,

I performed a test. See attachment.

[2024-11-28-10.29-Test-Opening-Veracrypt-container-with-wrong-password.pdf](https://github.com/user-attachments/files/17946288/2024-11-28-10.29-Test-Opening-Veracrypt-container-with-wrong-password.pdf)

Note: this issue is NO problem to me, I posted this issue to help you. I'm sorry if I offended you.

I am very happy with zulucrypt, because it is light weight, and also installs quickly in ramdisk if you boot from iso. Veracrypt is a heavy weight: ramdisk is almost full after installation.

WKR, Odie
